index.js

```diff
@@ -11,11 +12,46 @@ server.get('/payments', (req, res) => {
     res.status(200).send(payments);
 });
 
+server.get('/payments/:id', (req, res) => {
+    const payment = payments.find( payment => payment.id === req.params.id);
+    if(payment){
+        res.status(200).send(payment);
+    }
+    else{
+        res.status(404).send({message: 'Wrong id'});
+    }
+});
+
+server.post('/payments', (req, res) => {
+    const payment = {...req.body};
+    Object.assign(payment, {
+        id: uuidv4(),
+        status: 'Pending'
+    });
+    payments.push(payment);
+    res.status(200).send(payment);
+});
 
+server.put('/payments/:id', (req, res) => {
+    const payment = payments.find( payment => payment.id === req.params.id);
+    if(req.body.description){
+        payment.description = req.body.description;
```

Comment:
what if payment with the id provided wasn't found?
here is the model of the situation: 
```js
const payments = [
    {id: "111"},
    {id: "222"}
];

const payment = payments.find(payment => payment.id === "333");

// what is `payment `?
// what's gonna happen here: 
payment.description = "blah";
```